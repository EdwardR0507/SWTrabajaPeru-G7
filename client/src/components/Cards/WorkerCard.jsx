import React from 'react';
import { withStyles } from '@material-ui/core';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import IconButton from '@material-ui/core/IconButton';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import StarIcon from '@material-ui/icons/Star';
import StarHalfIcon from '@material-ui/icons/StarHalf';
import Typography from '@material-ui/core/Typography';

const StyledCard = withStyles({
    root:{
        width: '25%',
        margin: '20px auto'
    }
})(Card);

const StyledCardMedia = withStyles({
    root:{
        height: '0',
        paddingTop: '41%'
    }
})(CardMedia);


const WorkerCard = () => {
    return(
    <StyledCard>
        <StyledCardMedia 
            image="src/assets/CardTest.jpeg"
        />
        <CardContent>
            <Typography
                variant="h6">
                Nombre Usuario
            </Typography>
            <Typography
                variant="body2">
                Distrito/Provincia/Ciudad
            </Typography>
        </CardContent>
        <CardActions>
            <IconButton>
                <StarBorderIcon 
                    fontSize="small" style={{padding: 0}}/>
            </IconButton>
            <IconButton>
                <StarBorderIcon 
                    fontSize="small"style={{padding: 0}}/>
            </IconButton>
            <IconButton>
                <StarBorderIcon 
                    fontSize="small"/>
            </IconButton>
            <IconButton>
                <StarBorderIcon 
                    fontSize="small"/>
            </IconButton>
            <IconButton>
                <StarBorderIcon 
                    fontSize="small"/>
            </IconButton>
        </CardActions>
    </StyledCard>
    )
}

export default WorkerCard;