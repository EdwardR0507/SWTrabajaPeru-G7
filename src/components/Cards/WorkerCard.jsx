import React, { useState, useEffect } from "react";
import { withStyles } from "@material-ui/core";
import Card from "@material-ui/core/Card";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import Rating from "@material-ui/lab/Rating";
import Typography from "@material-ui/core/Typography";

const StyledCard = withStyles({
  root: {
    width: "300px",
    margin: "20px auto",
  },
})(Card);

const StyledCardMedia = withStyles({
  root: {
    height: "0",
    paddingTop: "41%",
    marginBottom: "10%",
  },
})(CardMedia);

const WorkerCard = (props) => {
  const [worker, setWorker] = useState({});

  useEffect(() => {
    setWorker(props.worker);
  }, [props.worker]);

  return worker ? (
    <StyledCard>
      <StyledCardMedia image="src/assets/CardTest.jpeg" />
      <CardContent>
        <Typography variant="h6">{worker.us_nombres}</Typography>
        <Typography variant="body2">
          {worker.us_distrito}/{worker.us_provincia}/{worker.us_departamento}
        </Typography>
      </CardContent>
      <CardActions>
        {worker.calificacion ? (
          <Rating name="read-only" value={worker.calificacion} readOnly />
        ) : (
          <Typography variant="body2">Sin Calificaci√≥n</Typography>
        )}
      </CardActions>
    </StyledCard>
  ) : (
    <>
      {console.log(worker)}
      <div>Cargando... </div>
    </>
  );
};

export default WorkerCard;
