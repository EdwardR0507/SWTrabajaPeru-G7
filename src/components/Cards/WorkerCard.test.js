/*Importamos las librerias principales*/
import React from "react";
import { shallow } from "enzyme";
import { render, screen } from '@testing-library/react'
import WorkerCard from "./WorkerCard";
/*Renderizado del componente WorkerCard*/
describe("<WorkerCard></WorkerCard>", () => {
  it('render "WorkerCard"', () => {
    const wrapper = shallow(<WorkerCard />);
    expect(wrapper).toMatchSnapshot()
  });
  it('render WorkerCard with user', () => {
    const user = {
      us_nombres: "Test Name",
      us_celular: "987654321",
    }
    render(<WorkerCard user={user} />);
    expect(screen.getByText(/Test Name/i)).toBeInTheDocument()
  })
});
