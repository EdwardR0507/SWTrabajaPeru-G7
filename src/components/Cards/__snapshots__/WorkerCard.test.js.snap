// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkerCard></WorkerCard> render "WorkerCard" 1`] = `ShallowWrapper {}`;
